generator/booking_dataset.cpp

```diff
@@ -136,28 +136,30 @@ bool BookingDataset::CanBeBooking(FeatureBuilder1 const & fb) const
   return ftypes::IsHotelChecker::Instance()(fb.GetTypes());
 }
 
-BookingDataset::Hotel const & BookingDataset::GetHotel(size_t index) const
+BookingDataset::Hotel const & BookingDataset::GetHotelById(uint32_t const id) const
 {
-  ASSERT_LESS(index, m_hotels.size(), ());
-  return m_hotels[index];
+  auto const it = m_hotels.find(id);
+  ASSERT(it != end(m_hotels), ());
```

Comment:
It would be helpful to write id in the error message. The same for the method below.